### PR TITLE
chore: add husky and lint-staged

### PR DESCRIPTION
**Pull request by jkodu**
This PR does the below:
- adds `husky`.
- adds `lint-staged`.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -180,5 +182,16 @@
 			"<rootDir>/.cache/",
 			"<rootDir>/.public/"
 		]
+	},
+	"husky": {
+		"hooks": {
+			"pre-commit": "npm test"
```
Comment (WilcoFiers) [inline package.json]: Actually, no I don't think we should run tests before a commit. We have CI for that.

Comment (jkodu) [reply package.json]: I know, but it will help people like me :).

Comment (WilcoFiers) [reply package.json]: > IF machine==jey npm test

Comment (WilcoFiers) [reply package.json]: Fine, I'll try it. If it gets on my nerves, I'm ripping it out though :P

**Pull request merged**